src/Microsoft.AspNet.Mvc.ViewFeatures/ViewExecutor.cs

```diff
@@ -37,32 +36,17 @@ public static class ViewExecutor
         {
             var response = actionContext.HttpContext.Response;
 
-            var contentTypeHeader = contentType;
-            Encoding encoding;
-            if (contentTypeHeader == null)
+            contentType = contentType ?? DefaultContentType;
+            if (contentType.Encoding == null)
             {
-                contentTypeHeader = DefaultContentType;
-                encoding = Encoding.UTF8;
-            }
-            else
-            {
-                if (contentTypeHeader.Encoding == null)
-                {
-                    // Do not modify the user supplied content type, so copy it instead
-                    contentTypeHeader = contentTypeHeader.Copy();
-                    contentTypeHeader.Encoding = Encoding.UTF8;
-
-                    encoding = Encoding.UTF8;
-                }
-                else
-                {
-                    encoding = contentTypeHeader.Encoding;
-                }
+                // Do not modify the user supplied content type, so copy it instead
+                contentType = contentType.Copy();
+                contentType.Encoding = Encoding.UTF8;
```

Comment:
I cleaned this up because I needed to copy the boilerplate required to create a writer. I didn't like the idea of copying like 20 loc, and then realized it was more like 4 so I felt better about it :laughing: